sphinx/themes/basic/layout.html

```diff
@@ -139,6 +139,9 @@
     {%- block scripts %}
     {{- script() }}
     {%- endblock %}
+    {%- if theme_canonical_url %}
+    <link rel="canonical" href="{{ theme_canonical_url }}{{ pagename }}.html"/>
```

Comment:
We can change the suffix of the HTML files through `html_file_suffix`. So we have to use variable to represent the name of file extension.